service/raft/raft_group_registry.cc

```diff
@@ -408,6 +424,82 @@ shared_ptr<raft::failure_detector> raft_group_registry::failure_detector() {
 
 raft_group_registry::~raft_group_registry() = default;
 
+namespace {
+    auto fmt_loc(const seastar::compat::source_location& l) {
+        return fmt::format("{}({}:{}) `{}`", l.file_name(), l.line(), l.column(), l.function_name());
+    };
+}
+
+raft_server_with_timeouts::raft_server_with_timeouts(raft_server_for_group& group_server, raft_group_registry& registry)
+    : _group_server(group_server)
+    , _registry(registry)
+{
+}
+
+template <std::invocable<abort_source*> Op>
+std::invoke_result_t<Op, abort_source*>
+raft_server_with_timeouts::run_with_timeout(Op&& op, const char* op_name,
+    seastar::abort_source* as, std::optional<raft_timeout> timeout)
+{
+    if (!timeout) {
+        co_await op(as);
+        co_return;
+    }
+    if (!timeout->value) {
+        if (!_group_server.default_op_timeout) {
```

Comment:
Can't we get rid of this default parameter and instead have a default for `raft_timeout::value`?

Reply:
The idea was to allow different timeouts for different groups of rafts.

Reply:
I see. I think group0 is a special one that can use global default timeout. Groups for consistent tables will have different timeouts for different kind of requests, so global setting will be less useful for them. But OK.